src/CakeSlayer/Measure.hs

```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE BangPatterns #-}
+
+{- | This module contains the 'MonadMeasure' type class which purpose is to
+apply some metric calculations on actions, e.g. taken time.
+-}
+
+module CakeSlayer.Measure
+       ( MonadMeasure
+       , timedAction
+
+         -- * Internals
+       , timedActionImpl
+       ) where
+
+import Prometheus (Histogram, Info (..), MonadMonitor, defaultBuckets, histogram, observe, register)
+import Relude.Extra.CallStack (ownName)
+import System.CPUTime (getCPUTime)
+import System.Metrics.Distribution (Distribution)
+
+import CakeSlayer.Has (Has, grab)
+
+import qualified Data.HashMap.Strict as HashMap
+import qualified Data.Map as Map
+import qualified System.Metrics as Metrics
+import qualified System.Metrics.Distribution as Distribution
+
+{- | This type class is used to perform action and measure time and other
+metrics needed to be performed.
+-}
+class Monad m => MonadTimed m where
+    timedAction :: HasCallStack => m a -> m a
+
+
+-- QUESTION: move to CakeSlayer.Env ?
+type Timings = IORef (Map Text Distribution)
+type PrometheusRegistry = IORef (HashMap Text Histogram)
```

Comment:
I think it's fine to have them here. Btw, what do you think about converting them to `newtype`s?

Reply:
I'm not sure. What are the benefits? I think that for these particular types it is not necessary.

Reply:
@vrom911 Okay, I agree. Maybe if in future there will be overlap or some problem with instances, we can convert to `newtype`s. For now, this is ok :+1: